File: task3_analyse_data/code/merge_yaml_stats.py
#!/usr/bin/env python3
"""
Merge the three level-specific YAML files into one combined file
"""

import yaml
from pathlib import Path

def main():
    base_path = Path("../graphs_and_maps/keyword_centroid")
    
    all_stats = {}
    
    # Load statistics for each level
    for level in ['sentence', 'article', 'section']:
        yaml_path = base_path / f"{level}_level" / "keyword_statistics.yaml"
        
        if yaml_path.exists():
            with open(yaml_path, 'r') as f:
                level_stats = yaml.safe_load(f)
                # Merge into main dict
                all_stats.update(level_stats)
            print(f"Loaded {len(level_stats)} statistics from {level} level")
        else:
            print(f"Warning: {yaml_path} not found")
    
    # Save combined YAML
    output_path = base_path / "all_keyword_statistics.yaml"
    with open(output_path, 'w') as f:
        yaml.dump(all_stats, f, default_flow_style=False, sort_keys=True)
    
    print(f"\nCombined statistics saved to: {output_path}")
    print(f"Total statistics: {len(all_stats)}")
    
    # Also save to manuscripts directory for easier access
    manuscripts_path = Path("../../manuscripts/keyword_statistics.yaml")
    with open(manuscripts_path, 'w') as f:
        yaml.dump(all_stats, f, default_flow_style=False, sort_keys=True)
    print(f"Also saved to: {manuscripts_path}")

if __name__ == "__main__":
    main()

File: task3_analyse_data/code/merge_yaml_stats.R
#!/usr/bin/env Rscript
# Merge the three level-specific YAML files into one combined file

library(yaml)

# Function to merge YAML files
merge_yaml_stats <- function() {
    base_path <- "../graphs_and_maps/keyword_centroid"
    
    all_stats <- list()
    
    # Load statistics for each level
    for (level in c('sentence', 'article', 'section')) {
        yaml_path <- file.path(base_path, paste0(level, "_level"), "keyword_statistics.yaml")
        
        if (file.exists(yaml_path)) {
            level_stats <- read_yaml(yaml_path)
            # Merge into main list
            all_stats <- c(all_stats, level_stats)
            cat(sprintf("Loaded %d statistics from %s level\n", length(level_stats), level))
        } else {
            cat(sprintf("Warning: %s not found\n", yaml_path))
        }
    }
    
    # Save combined YAML
    output_path <- file.path(base_path, "all_keyword_statistics.yaml")
    write_yaml(all_stats, output_path)
    cat(sprintf("\nCombined statistics saved to: %s\n", output_path))
    cat(sprintf("Total statistics: %d\n", length(all_stats)))
    
    # Also save to manuscripts directory for easier access
    manuscripts_path <- "../../manuscripts/keyword_statistics.yaml"
    write_yaml(all_stats, manuscripts_path)
    cat(sprintf("Also saved to: %s\n", manuscripts_path))
}

# Run the merge
merge_yaml_stats()